src/completion-listener.ts

```diff
@@ -86,10 +86,18 @@ export class CompletionListener {
      * @returns A Promise that resolves if the success condition is met, or rejects otherwise.
      */
     listen(commands: Command[]): Promise<CloseEvent[]> {
-        const closeStreams = commands.map((command) => command.close);
+        const closeStreams = commands.map((command) => command.close.pipe());
+
         return Rx.lastValueFrom(
-            Rx.merge(...closeStreams).pipe(
-                bufferCount(closeStreams.length),
+            Rx.combineLatest(closeStreams).pipe(
+                filter((exitInfos) => exitInfos.every((exitInfo) => exitInfo.exitCode !== null)),
```

Comment:
`exitCode` is never null here. You need to look at the event's command for the current state

```suggestion
                filter((exitInfos) => exitInfos.every(({ command }) => command.state !== 'started')),
```

Reply:
Yea good point 👍 It would mean we will change the `CommandInfo` interface to include it as well 👌

Reply:
Heya, I've fixed this myself and added an extra test too.

You were right, adding to `CommandInfo` is not a good idea, but so wasn't right to pass the state in the close event since that's pretty static. Can check `commands` variable just outside the loop, though.